src/requests/GroupsApi.ts

```diff
@@ -0,0 +1,1246 @@
+
+        if (joinState) {
+            if (!Object.values(GroupJoinState).includes(joinState)) {
+                console.warn('JoinState must be one of the following: 0 - Closed, 1 - Invite, 2 - Request, 3 - Open! Parameter ommited! Defaulted to "0 - Open"');
+            } else {
+                body.joinState = joinState;
+            }
+        }
+        
+        if (iconId) {
+            body.iconId = iconId;
+        }
+
+        if (bannerId) {
+            body.bannerId = bannerId;
+        }
+
+        if (privacy) {
+            if (!Object.values(GroupPrivacy).includes(privacy)) {
+                console.warn('Privacy must be one of the following: 0 - Default, 1 - Private! Parameter ommited! Defaulted to "0 - Default"!');
+            } else {
+                body.privacy = privacy;
+            }
+        }
+
+        const paramRequest: VRCAPI.Generics.executeRequestType = {
+            currentRequest: ApiPaths.groups.createGroup,
+            pathFormated: ApiPaths.groups.createGroup.path,
+            body: body,
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Groups.Models.Group>(paramRequest);
+
+        return queryResult;
+
+    }
+
+    /**
+     * Returns a single Group by ID.
+     * @param param0 
+     * @returns 
+     */
+    public async getGroupbyID({ groupId, includeRoles = false }: VRCAPI.Groups.Requests.getGroupByIdRequest): Promise<VRCAPI.Groups.Models.Group> {
+        const parameters: URLSearchParams = new URLSearchParams();
+
+        if (!groupId) {
+            throw new Error('No group id was provided!');
+        }
+
+        parameters.append('includeRoles', includeRoles.toString());
+
+        const paramRequest: VRCAPI.Generics.executeRequestType = {
+            currentRequest: ApiPaths.groups.getGroupById,
+            pathFormated: ApiPaths.groups.getGroupById.path.replace('{groupId}', groupId),
+            queryOptions: parameters,
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Groups.Models.Group>(paramRequest);
+
+        return queryResult;
+    }
+
+    /**
+     * Updates a Group and returns it.
+     * @param param0 
+     * @returns 
+     */
+    public async updateGroup( {
+        groupId,
+        data: {
+          name,
+          shortCode, // 3 to 6 chars
+          description, // Must be 0 to 250 characters long, optional
+          joinState,
+          iconId,
+          bannerId,
+          languages, // max length of 3, each max 3 characters strings
+          links, // max length of 3
+          rules, // max 2048 characters
+        }
+      } : VRCAPI.Groups.Requests.UpdateGroupRequest): Promise<VRCAPI.Groups.Models.Group> {
+
+        // at least one parameter of the data must be present to continue
+        if (!name && !shortCode && !description && !joinState && !iconId && !bannerId && !languages && !links && !rules) {
```

Comment:
Optional: There's a way to rewrite this better, this works, but it's a challenge for you to try to make this differently.